pkg/storage/stores/indexshipper/compactor/table.go

```diff
@@ -109,6 +112,7 @@ func newTable(ctx context.Context, workingDirectory string, indexStorageClient s
 		baseUserIndexSet:   storage.NewIndexSet(indexStorageClient, true),
 		baseCommonIndexSet: storage.NewIndexSet(indexStorageClient, false),
 		uploadConcurrency:  uploadConcurrency,
+		parallelism:        parallelism,
```

Comment:
Is there a conceptual difference between `uploadConcurrency` and `parallelism`, except that the former is for uploading tables, the latter is for downloading tables?

Maybe the settings could be consolidated?

Reply:
I considered it, but if I remove `uploadConcurrency` it would be a breaking change, if I reuse `uploadConcurrency` everywhere it would be misleading and can lead to unexpected behavior, as everyone using low `uploadConcurrency` would experience a degradation in parallelism on other parts of the code.